solutions/src/main/java/ru/sberbank/school/task10/MyScalableThreadPool.java

```diff
@@ -0,0 +1,189 @@
+            tasks.add(futureTask);
+            tasks.notifyAll();
+            return futureTask;
+        }
+    }
+
+    /*
+    *return true if needed to interrupt, otherwise return false
+    *  */
+    private boolean poolResize () {
+        int taskSize = tasks.size();
+
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> {
+
+                if (worker == null || worker.isInterrupted()) {
+                    worker = null;
+                    if (!freeIndexForNewWorker.contains(key)) {
+                        freeIndexForNewWorker.offer(key);
+                    }
+
+                    synchronized (activeThreadsLock) {
+                        activeThreads--;
+                    }
+                }
+            });
+
+            if (taskSize > sizeMin && activeThreads < sizeMax) {
+                synchronized (activeThreadsLock) {
+                    activeThreads++;
+                }
+
+                if (!freeIndexForNewWorker.isEmpty()) {
+                    int id = freeIndexForNewWorker.poll();
+
+                    ThreadWorker worker =new ThreadWorker("ThreadWorker " + id, id);
+                    poolMap.put(id, worker);
+                    worker.start();
+                }
+            }
+
+            return taskSize < activeThreads && activeThreads > sizeMin;
+        }
+    }
+
+    public int getActiveThreads () {
+        return activeThreads;
+    }
+
+    private class ThreadWorker extends Thread {
+        private int id;
+
+        private ThreadWorker(String name, int id) {
+            super(name);
+            this.id = id;
+        }
+
+        @Override
+        public void run() {
+            System.out.println("ThreadWorker "+ id + " started");
+            while (!Thread.currentThread().isInterrupted()) {
+
+                synchronized (tasks) {
+                    while (tasks.isEmpty()) {
+                        try {
+                            tasks.wait();
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                            break;
+                        }
+                    }
+
+                    if (!Thread.currentThread().isInterrupted()) {
+                        boolean needToInterrupt = poolResize();
+
+                        if (needToInterrupt) {
+                            synchronized (activeThreadsLock) {
+                                activeThreads--;
+                            }
+                            System.out.println("ThreadWorker "+ id + " interrupted");
+                            Thread.currentThread().interrupt();
+                            break;
+                        }
+
+                        FutureTask task = tasks.poll();
+
+                        if (task != null) {
+                            task.run();
+                            System.out.println("ThreadWorker "+ id + " ended");
```

Comment:
WhitespaceAround: &#39;+&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1963080994>*